drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -578,4 +593,47 @@ def introspect_custom_data_element_definitions(set_definition_identifier: false)
 
     cded_records
   end
+
+  def set_hud_requirements
+    changed = [] # list of Link IDs that were changed
+    rule_module = HmisUtil::HudAssessmentFormRules2024.new
+
+    walk_definition_nodes do |item|
+      link_id = item['link_id']
+
+      # Check if there is a required data collection rule for this link_id for this data collection stage (ie role)
+      hud_rule = rule_module.hud_data_element_rule(role, link_id)
+      if hud_rule
+        # If the rule is different from what's on the item, update it
+        difference = Hashdiff.diff(hud_rule, item['rule'], ignore_keys: '_comment')
```

Comment:
Recommend adding the hashdiff gem as an explicit dependency to the Gemfile. It's in the project already but only because it's a secondary dependency of another gem.